From acedc9246ae68a9da0f72eb48fdc1027dc50c36c Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Mon, 9 May 2016 17:55:30 -0400
Subject: [PATCH 1555/1722] drm/amdgpu: don't add_bo_to_vm in import_dmabuf

The BO is added to the VM with the proper BO and VM reservations in
map_memory_to_gpu. This fixes several kernel warnings from TTM during
import_dmabuf.

Change-Id: Ie92c6a22061d860e7257f3e2b86e2b2d97362f2c
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 4326a51..2ba6592 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -1397,7 +1397,7 @@ int amdgpu_amdkfd_gpuvm_import_dmabuf(struct kgd_dev *kgd, int dma_buf_fd,
 	struct dma_buf *dma_buf;
 	struct drm_gem_object *obj;
 	struct amdgpu_bo *bo;
-	int r = -EINVAL;
+	int r = 0;
 
 	dma_buf = dma_buf_get(dma_buf_fd);
 	if (IS_ERR(dma_buf))
@@ -1438,14 +1438,6 @@ int amdgpu_amdkfd_gpuvm_import_dmabuf(struct kgd_dev *kgd, int dma_buf_fd,
 		AMDGPU_GEM_DOMAIN_VRAM : AMDGPU_GEM_DOMAIN_GTT;
 	(*mem)->data2.mapped_to_gpu_memory = 0;
 
-	r = add_bo_to_vm(adev, *mem, vm, false, NULL);
-
-	if (r) {
-		amdgpu_bo_unref(&bo);
-		kfree(*mem);
-		*mem = NULL;
-	}
-
 out_put:
 	dma_buf_put(dma_buf);
 	return r;
-- 
2.7.4

